### PR TITLE
added logging for prep_start; prep_start now returns the appropriate …

### PR DESCRIPTION
**Pull request by cprafullchandra**
…tuple of success, value

### CODE REVIEW

File: vent/api/actions.py [hunk 1]
```diff
@@ -79,221 +83,272 @@ def prep_start(self,
...
-            os.chdir(cwd)
-
-            if run_build:
-                status = self.build(name=sections[section]['name'],
-                                    groups=groups,
-                                    enabled=enabled,
-                                    branch=branch,
-                                    version=version)
+        try:
+            args = locals()
```
Comment (cglewis) [inline vent/api/actions.py]: This has to be the first thing in the function.

File: vent/api/actions.py [hunk 2]
```diff
@@ -79,221 +83,272 @@ def prep_start(self,
...
+                       'built',
+                       'groups',
+                       'path',
+                       'image_name',
+                       'branch',
+                       'version']
+            vent_config = Template(template=self.vent_config)
+            files = vent_config.option('main', 'files')
+            sections, template = self.plugin.constraint_options(args, options)
+            self.logger.info(sections)
```
Comment (cglewis) [inline vent/api/actions.py]: this seems excessive.

File: vent/api/actions.py [hunk 3]
```diff
@@ -79,221 +83,272 @@ def prep_start(self,
...
-            tool_dict[container_name]['labels']['vent.name'] = sections[section]['name']
-
-            if 'groups' in sections[section]:
-                # add labels for groups
-                tool_dict[container_name]['labels']['vent.groups'] = sections[section]['groups']
-                # send logs to syslog
-                if 'syslog' not in sections[section]['groups'] and 'core' in sections[section]['groups']:
-                    tool_dict[container_name]['log_config'] = {'type':'syslog', 'config': {'syslog-address':'tcp://0.0.0.0:514', 'syslog-facility':'daemon', 'tag':'core'}}
-                if 'syslog' not in sections[section]['groups']:
+                self.logger.info(plugin_sections)
```
Comment (cglewis) [inline vent/api/actions.py]: excessive

File: vent/api/actions.py [hunk 4]
```diff
@@ -79,221 +83,272 @@ def prep_start(self,
...
+                # add extra labels
+                if 'labels' not in tool_dict[container_name]:
+                    tool_dict[container_name]['labels'] = {}
+                tool_dict[container_name]['labels']['vent'] = Version()
+                tool_dict[container_name]['labels']['vent.namespace'] = sections[section]['namespace']
+                tool_dict[container_name]['labels']['vent.branch'] = branch
+                tool_dict[container_name]['labels']['vent.version'] = version
+                tool_dict[container_name]['labels']['vent.name'] = sections[section]['name']
+
+                self.logger.info(tool_dict)
```
Comment (cglewis) [inline vent/api/actions.py]: why twice?

File: vent/api/actions.py [hunk 5]
```diff
@@ -79,221 +83,272 @@ def prep_start(self,
...
-                del tool_dict[c]['tmp_name']
-
-        # remove containers that shouldn't be started
-        for c in tool_dict.keys():
-            if 'start' in tool_dict[c] and not tool_dict[c]['start']:
-                del tool_dict[c]
-
-        return tool_dict
+
+                self.logger.info(tool_dict)
```
Comment (cglewis) [inline vent/api/actions.py]: three times...

File: vent/api/actions.py [hunk 6]
```diff
@@ -79,221 +83,272 @@ def prep_start(self,
...
+                if status[0]:
+                    for option in status[1]:
+                        if option[0] == 'priority':
+                            tool_dict[container_name]['labels']['vent.priority'] = option[1]
+
+                # only start tools that have been built
+                if sections[section]['built'] != 'yes':
+                    del tool_dict[container_name]
+
+                self.logger.info(tool_dict)
```
Comment (cglewis) [inline vent/api/actions.py]: and more...

File: vent/api/actions.py [hunk 7]
```diff
@@ -79,221 +83,272 @@ def prep_start(self,
...
+                                tmp_volumes_from.remove(volumes_from)
+                    tool_dict[container]['volumes_from'] += tmp_volumes_from
+                if 'network_mode' in tool_dict[container]:
+                    if tool_dict[container]['network_mode'].startswith('container:'):
+                        network_c_name = tool_dict[container]['network_mode'].split('container:')[1]
+                        for c in tool_dict.keys():
+                            if 'tmp_name' in tool_dict[c] and tool_dict[c]['tmp_name'] == network_c_name:
+                                tool_dict[container]['network_mode'] = 'container:'+tool_dict[c]['name']
+
+            self.logger.info(tool_dict)
```
Comment (cglewis) [inline vent/api/actions.py]: and more...

File: vent/api/actions.py [hunk 8]
```diff
@@ -79,221 +83,272 @@ def prep_start(self,
...
+        except Exception as e:
+            self.logger.error(str(e))
+            status = (False, e)
+
+        status = (True, tool_dict)
+        self.logger.info(status)
+        self.logger.info("Finished: prep_start")
+        return status
```
Comment (cglewis) [inline vent/api/actions.py]: how does start get `tool_dict` if you return `status`?

**Comment by cglewis** [comment]
@cprafullchandra think a bit about what you're logging and what the end user (not developer) will see.  make the output useful for them, but just a bunch of crazy output.  add a bit of prose in front of the variable you're logging to let them know what it is that they're looking at.  don't log things that they don't need to care about from a informational perspective.

File: vent/api/actions.py (hunk 8 above)
Comment (cprafullchandra) [inline vent/api/actions.py]: I think I've modified start accordingly so that it takes status[1] to get tool_dict for the update that it performs.

Comment (cglewis) [inline vent/api/actions.py]: ah, sorry missed that. cool.

**Pull request merged**